lib/puppet/indirector/file_metadata/http.rb

```diff
@@ -18,8 +18,8 @@ def find(request)
 
     case head.code
     when 403
-      # AMZ presigned URL?
-      if head.each_header.find { |k,_| k =~ /^x-amz-/i }
+      # If AMZ presigned URL or puppetserver, fallback to partial get
+      if head.each_header.find { |k,_| k =~ /^(x-amz-|x-puppet-)/i }
```

Comment:
Wondering if maybe we should always fallback to partial GET?

Reply:
Thoughts @GabrielNagy ^?

Reply:
Sounds good to me :+1:

Reply:
Updated